### ISSUE TITLE
Autocomplete for columns

### ISSUE THREAD

**Issue by AndreasTA-AW** [innoverio/vscode-dbt-power-user#79]
Hi there!

I don't think this plugin supports autocompletion for columns as of today. 
Would this be something that would be possible to add? 

It should be possible to read the .yml files, and if the columns exists, give some guidance about different columns. Could be even more advanced in the future and actually find columns from sql files and such, but that might be overkill to start with :P

Cheers!

**Comment by mdesmet** [comment]
It's not supported indeed. If I understand correctly you would like autocomplete of columns while writing a query?

I agree it would be a really nice feature but I think it's pretty tricky to implement properly. Ideally you would get meta data from the database (the easy part) but on top of that you would also have to parse the query to know what are the relevant columns within that query (think about joins, aliasing, ..., this is the hard part).

**Comment by jjuanramos** [comment]
I think this could be really interesting to implement by taking advantage of dbt documentation.
You could be able to autocomplete columns of a given model only when those columns have been documented on a given YAML file.

This also would encourage best practices in regards to documentation, immediately rewarding users for documenting the columns they use/create.

What do you think?

**Comment by AndreasTA-AW** [comment]
Yeah, we will most likely always document the columns anyway, so that should be fine. It's even possible to autogenerate that kind of documentation for sources and such.

**Comment by acjay** [comment]
I think @mdesmet is saying that when writing the CompletionItemProvider, you would have to be able to detect that you're in position where a column name is expected, and if so, determine which column names to present as choices based on the table. In SQL, you don't need to specify the table from the FROM or JOIN clauses if its not ambiguous, so the extension would have to be able to determine that context. Also, it would have to resolve aliases. DBT's macros for referencing tables might make this a easier in some ways (getting the column names from the manifest) and harder in others (can't use an off-the-shelf SQL parser to determine context).

**Comment by acjay** [comment]
Maybe as a V1, assume this only works for column references that have a table name or alias, and therefore can be detected by the dot character (with care to filter out floating point numbers, dots in quotes, dots for schema names, etc).

**Comment by aldencabajar** [comment]
This is definitely something that should be added and is doable by referencing the yaml files. Any work being done to this?

**Comment by mdesmet** [comment]
AFAIK no one is working on this. Feel free to start working on this.

**Comment by aldencabajar** [comment]
> AFAIK no one is working on this. Feel free to start working on this.

I'll take a stab at this. Will let you know once I've had some substantial work.

**Comment by mdesmet** [comment]
Contact me on dbt Slack if you need any help!

**Comment by krutoileshii** [comment]
is this still being looked at?
